streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueLoggedStore.java

```diff
@@ -79,6 +79,15 @@ public void put(K key, V value) {
     }
 
     @Override
+    public void putIfAbsent(K key, V value) {
+        if (get(key) == null) {
+            this.inner.putIfAbsent(key, value);
+            changeLogger.add(key);
+            changeLogger.maybeLogChange(this.getter);
+        }
```

Comment:
Yes, I would probably be best to follow use the same signature here, I'll change this